src/libServer/ScillaIPCServer.cpp

```diff
@@ -245,14 +245,18 @@ bool ScillaIPCServer::fetchBlockchainInfo(const std::string &query_name,
   // block.
   blockNum = m_BCInfo->getCurDSBlockNum();
   DSBlockSharedPtr dsBlockSharedPtr;
-  if (query_name == "BLOCKCOINBASE" || query_name == "BLOCKDIFFICULTY" ||
+  if (query_name == "BLOCKHASH" || query_name == "BLOCKCOINBASE" || query_name == "BLOCKDIFFICULTY" ||
       query_name == "BLOCKGASPRICE") {
     if (!BlockStorage::GetBlockStorage().GetDSBlock(blockNum,
                                                     dsBlockSharedPtr)) {
       LOG_GENERAL(WARNING, "Could not get blockNum DS block " << blockNum);
       return false;
     }
   }
+  if (not dsBlockSharedPtr || not txBlockSharedPtr) {
+    LOG_GENERAL(WARNING, "Smart pointers work better when Initialized ");
```

Comment:
:)

Reply:
I will make that more sensible before closing.